pkg/pgxconn/pgx_conn.go

```diff
@@ -13,9 +13,47 @@ import (
 	"github.com/jackc/pgconn"
 	"github.com/jackc/pgx/v4"
 	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/prometheus/client_golang/prometheus"
+
 	"github.com/timescale/promscale/pkg/log"
+	"github.com/timescale/promscale/pkg/util"
+)
+
+var (
+	requestTotal = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: util.PromNamespace,
+			Subsystem: "database",
+			Name:      "requests_total",
+			Help:      "Total number of database requests.",
+		}, []string{"method"},
+	)
+	errorsTotal = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: util.PromNamespace,
+			Subsystem: "database",
+			Name:      "request_errors_total",
+			Help:      "Total number of database request errors.",
+		}, []string{"method"},
+	)
+	duration = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: util.PromNamespace,
+			Subsystem: "database",
+			Name:      "requests_duration_seconds",
+			Help:      "Time taken to complete a database request.",
+		}, []string{"method"},
+	)
 )
 
+func init() {
+	prometheus.MustRegister(requestTotal, errorsTotal, duration)
+}
+
+func getLabelSet(method string) prometheus.Labels {
```

Comment:
Nit: I am often wondering if it's worth replacing one liner with another one liner :) Maybe consider naming function: `promMethodLabel` ?